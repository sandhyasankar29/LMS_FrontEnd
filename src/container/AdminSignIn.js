import { React, useState } from 'react';
import axios from 'axios';
import { useHistory } from "react-router-dom";
import './GlobalVariables'  
import {
    Box,
    Button,
    Grommet,
    TextInput,
    Image,
} from 'grommet';
import { Hide, View, Home } from 'grommet-icons';
import signImage from '../assets/adminSignIn.svg'
import { grommet } from 'grommet/themes';
import Swal from 'sweetalert2'
const AdminSignIn = () => {
    const urlSignIn = 'http://'+{global,ip}.ip+':3001/signinadmin';
    console.log(urlSignIn);
    const history = useHistory();
    const [password, setPassword] = useState("")
    const [reveal, setReveal] = useState(false);
    const onSubmitSignIn = () => {
        if (password) {
            axios.post(urlSignIn, { password: password })
                .then(res => {
                    history.replace('/adminteacherverify')
                }).catch(error => {
                    Swal.fire({
                        icon: 'error',
                        title: 'Oops...',
                        text: 'Check your Credentials'
                    })
                })
        }
    }
    document.body.style.overflow = "hidden"
    return (
        <Grommet full theme={grommet}>
            <Box pad="small" justify="start" align="start" >
                <Button primary icon={<Home />} id="HomeBtn" label="Go to Home" hoverIndicator="light-1" onClick={() => { history.replace('/') }} />
            </Box>
            <Box fill align="center" justify="center">
                <Box height="small" width="large" style={{
                    marginBottom: "3%",
                }}>
                    <Image src={signImage} fit="contain" />
                </Box>

                <Box width="medium">
                    <Box
                        width="medium"
                        direction="row"
                        align="center"
                        round="small"
                        border
                    >
                        <TextInput
                            plain
                            placeholder="Enter Master Key"
                            name="password" type="password"
                            // eslint-disable-next-line react/jsx-no-duplicate-props
                            type={reveal ? 'text' : 'password'}
                            value={password}
                            onChange={(event) => setPassword(event.target.value)}
                        />
                        <Button
                            icon={reveal ? <View size="medium" /> : <Hide size="medium" />}
                            onClick={() => setReveal(!reveal)}
                        />
                    </Box>

                    <Box direction="row" justify="center" margin={{ top: 'medium' }}>
                        <Button data-testid="button" id="submitBtn" active={true} onClick={onSubmitSignIn} type="submit" label="Log In" primary />
                    </Box>
                </Box>
            </Box>
        </Grommet>
    );
};

export default AdminSignIn;