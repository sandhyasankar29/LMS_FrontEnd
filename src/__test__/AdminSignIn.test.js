
import React from 'react';
import ReactDOM from 'react-dom';
import AdminSignIn from '../container/AdminSignIn';
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

afterEach(cleanup)
it ("renders without crashing",()=>{
    const div=document.createElement("div");
    ReactDOM.render(<AdminSignIn></AdminSignIn>,div)
})

it("renders button correctly", () => {
    const { getByTestId } = render(<AdminSignIn></AdminSignIn>)
    expect(getByTestId('button')).toHaveTextContent("Log In")
});


it("matches snapshot", () => {
    const tree = renderer.create(<AdminSignIn></AdminSignIn>).toJSON();
    expect(tree).toMatchSnapshot();
})