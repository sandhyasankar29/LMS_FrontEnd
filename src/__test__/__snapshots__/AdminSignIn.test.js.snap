// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 bWDPZI"
>
  <div
    className="StyledBox-sc-13pk1d4-0 HvGCm"
  >
    <button
      className="StyledButton-sc-323bzc-0 bexXh"
      id="HomeBtn"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 iQEJEP"
      >
        <svg
          aria-label="Home"
          className="StyledIcon-ofa7kd-0 dcacLZ"
          viewBox="0 0 24 24"
        >
          <path
            d="M1,11 L12,2 L23,11 M15,23 L15,15 L15,15 L9,15 L9,23 M4,23 L4,9 M20,23 L20,9"
            fill="none"
            stroke="#000"
            strokeWidth="2"
          />
        </svg>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hjorfk"
        />
        Go to Home
      </div>
    </button>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 bxevCE"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 bJtzEt"
      style={
        Object {
          "marginBottom": "3%",
        }
      }
    >
      <img
        className="StyledImage-ey4zx9-0 iBeQsm"
        src="adminSignIn.svg"
      />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 gfocoz"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 clcAnq"
      >
        <div
          className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
        >
          <input
            autoComplete="off"
            className="StyledTextInput-sc-1x30a0s-0 hubMyb"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Enter Master Key"
            type="password"
            value=""
          />
        </div>
        <button
          className="StyledButton-sc-323bzc-0 hiyJPE"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          <svg
            aria-label="Hide"
            className="StyledIcon-ofa7kd-0 dEeaTZ"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,17 C9.27272727,17 6,14.2222222 6,12 C6,9.77777778 9.27272727,7 12,7 C14.7272727,7 18,9.77777778 18,12 C18,14.2222222 14.7272727,17 12,17 Z M11,12 C11,12.55225 11.44775,13 12,13 C12.55225,13 13,12.55225 13,12 C13,11.44775 12.55225,11 12,11 C11.44775,11 11,11.44775 11,12 Z M20,5 L4,19"
              fill="none"
              stroke="#000"
              strokeWidth="2"
            />
          </svg>
        </button>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 cgsQvy"
      >
        <button
          className="StyledButton-sc-323bzc-0 iVPikU"
          data-testid="button"
          id="submitBtn"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="submit"
        >
          Log In
        </button>
      </div>
    </div>
  </div>
</div>
`;
